client/components/SignUpBox.test.jsx

```diff
@@ -0,0 +1,115 @@
+import { shallow, mount } from "enzyme";
+import { render, fireEvent, getByTestId, cleanup } from "react-testing-library";
+import React from "react";
+import renderer from "react-test-renderer";
+
+import SignUpBox from "./SignUpBox";
+
+afterEach(cleanup);
+
+// Begin client-side tests
+describe("With Enzyme", () => {
+  it('SignUpBox shows "Sign Up"', () => {
+    const signup = shallow(<SignUpBox />);
+    expect(signup.find("h1").text()).toEqual("Sign Up");
+  });
+});
+
+describe("With Snapshot Testing", () => {
+  it('SignUpBox shows "Sign Up"', () => {
+    const component = renderer.create(<SignUpBox />);
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
+
+describe("SignUpBox elements", () => {
+  it("has an input field for email", () => {
+    const wrapper = mount(<SignUpBox />);
+    const emailInput = wrapper.find("#email");
+    expect(emailInput.exists()).toEqual(true);
+    expect(emailInput.matchesElement(<input id="email" />)).toEqual(true);
+  });
+
+  it("has an input field for password", () => {
+    const wrapper = mount(<SignUpBox />);
+    const passwordInput = wrapper.find("#password");
+    expect(passwordInput.exists()).toEqual(true);
+    expect(passwordInput.matchesElement(<input id="password" />)).toEqual(true);
+  });
+
+  it("has has a submit button", () => {
+    const wrapper = mount(<SignUpBox />);
+    const submitButton = wrapper.find("button");
+    expect(submitButton.exists()).toEqual(true);
+    expect(submitButton.text()).toEqual("Sign Up");
+  });
+});
+
+describe("Results of SignUpBox events", () => {
+  it("renders <p> tag and displays appropriate message", () => {
+    const wrapper = mount(<SignUpBox />);
+    const submitButton = wrapper.find("button");
+    submitButton.simulate("click");
+    expect(wrapper.find("p").exists()).toEqual(true);
+
+    // input form fields are completely empty
+    expect(wrapper.find("p").text()).toEqual(
+      "Password must be at least 6 characters in length"
+    );
+  });
+});
+
+test("SignUpBox loads with intitial state of '' ", () => {
+  const { container } = render(<SignUpBox />);
+  const emailVal = getByTestId(container, "emailVal");
```

Comment:
Can you get `emailVal` by just calling `.find("#email")`, instead of using `getByTestId`?

Reply:
No, you would have to use `const emailVal = mount(<SignUpBox />).find('#email'). That's the Enzyme way. I used the `data-testId` to keep it unique to react-testing-library.

Reply:
Ah I see. `react-testing-library` does not have an equivalent of `.find` in enzyme.